### PR TITLE
Custom error page when failing to load Supervisor panel

### LINKED ISSUE #6820: Better error message when Supervisor panel fails
## The request

The supervisor panel should not fail to load. But sometimes the supervisor stops responding and then users are presented with a generic custom panel error message.

When a custom panel fails to load, we should check if the panel that we're trying to load is the Supervisor panel. If so, we should provide a better error screen including a link to the website where we can help troubleshooting / telling you to reboot your device.

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #6820
- This PR is related to issue or discussion:
- Link to documentation pull request:
 
image

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/entrypoints/custom-panel.ts [hunk 1]
```diff
@@ -86,7 +88,49 @@ function initialize(
...
+          }
+          --mdc-theme-primary: var(--primary-color);
+          --header-height: 56px;
+          font-family: Roboto, sans-serif;
+          display: block;
+          background-color: var(--primary-background-color);
+          color: var(--primary-text-color);
+          height: calc(100vh - 32px);
+          width: 100vw;
+        }`;
```
Comment (bramkragten) [inline src/entrypoints/custom-panel.ts:129]: (I know it is a draft)

We should keep these in a const in a file and import them at the multiple places we use them, (onboarding, auth, here, etc)

File: src/entrypoints/custom-panel.ts [hunk 2]
```diff
@@ -86,7 +88,49 @@ function initialize(
       (err) => {
         // eslint-disable-next-line
         console.error(err, panel);
-        alert(`Unable to load the panel source: ${err}.`);
+        let errorScreen;
```
Comment (bramkragten) [inline src/entrypoints/custom-panel.ts:91]: We could also use `window.parent.customPanel.navigate` to get out of the iframe

Comment (ludeeus) [reply src/entrypoints/custom-panel.ts:91]: neat!

File: src/entrypoints/custom-panel.ts (hunk 1 above)
Comment (ludeeus) [reply src/entrypoints/custom-panel.ts:129]: yeah, just put them here to have a quick way to difference while testing

File: src/layouts/hass-error-screen.ts [hunk 3]
```diff
@@ -32,7 +32,9 @@ class HassErrorScreen extends LitElement {
       <div class="content">
         <h3>${this.error}</h3>
         <slot>
-          <mwc-button @click=${this._handleBack}>go back</mwc-button>
+          <mwc-button @click=${this._handleBack}>
+            ${this.hass.localize("ui.panel.error.go_back")}
```
Comment (ludeeus) [inline src/layouts/hass-error-screen.ts:36]:
```suggestion
            ${this.hass?.localize("ui.panel.error.go_back") || "go back"}
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**